torch/_subclasses/meta_utils.py

```diff
@@ -241,6 +241,18 @@ def sym_sizes_strides_storage_offset(
             t, src, symbolic_context=symbolic_context
         ) -> Tuple[Tuple[int, ...], Tuple[int, ...], int]:
             if shape_env is not None:
+                def metafy_fn(t, src) -> torch.Tensor:
+                    _symbolic_context = None
+                    if symbolic_context is not None:
+                        _symbolic_context = symbolic_context.tensor_to_inner_context[t]
```

Comment:
This doesn't actually work in this version of the data structure where we maintain a canonical nested int because the vec on the nested int won't necessarily the same vec as the offsets/lengths.

Reply:
One way to do this while keeping the current "canonical nested int" design is to maintain weak ref to all vec in the equiv set.

The alternative might be just to enforce that each vec must associated with its own nested int and enforce that NT's offset/lengths is always the same tensor as the vec in its nested int.

Reply:
I've updated the PR to do the "maintain a weak ref to every vec" approach. Still need to add a test.